src/qvs.cpp

```diff
@@ -0,0 +1,316 @@
+
+double QVSStls::computeAlpha() {
+  //  Compute the free energy integrand
+  thermoProp.compute<QVSStls>(in);
+  // Free energy
+  const vector<double> freeEnergyData = thermoProp.getFreeEnergyData();
+  const double& fxcr = freeEnergyData[1];
+  const double& fxcrr = freeEnergyData[2];
+  const double& fxct = freeEnergyData[3];
+  const double& fxctt = freeEnergyData[4];
+  const double& fxcrt = freeEnergyData[5];
+  // Q
+  const vector<double> QData = thermoProp.getQData();
+  const double& Q = QData[0];
+  const double& Qr = QData[1];
+  const double& Qt = QData[2];
+  // Alpha
+  double numer = Q - (1.0/6.0) * fxcrr + (1.0/3.0) * fxcr;
+  double denom =  Q + (1.0/3.0) * Qr;
+  if (in.getDegeneracy() > 0.0) {
+    numer += - (2.0/3.0) * fxctt
+             - (2.0/3.0) * fxcrt
+             + (1.0/3.0) * fxct;
+    denom += (2.0/3.0) * Qt;
+  }
+  return numer/denom;
+}
+
+void QVSStls::updateSolution() {
+  // Update the structural properties used for output
+  const auto& qstls = thermoProp.getStructProp<QStlsCSR>();
+  adr = qstls.getAdr();
+  ssf = qstls.getSsf();
+}
+
+// -----------------------------------------------------------------
+// qThermoProp class
+// -----------------------------------------------------------------
+
+vector<double> QThermoProp::getQData() const {
+  // QAdder
+  const std::vector<double> qVec = structProp.getQ();
+  const std::vector<double> rs = structProp.getCouplingParameters();
+  const double q = qVec[SIdx::RS_THETA] / rs[SIdx::RS_THETA];
+  // QAdder derivative with respect to the coupling parameter
+  double qr;
+  {
+    const double drs = rs[SIdx::RS_UP_THETA] - rs[SIdx::RS_THETA];
+    const double& q0 = qVec[SIdx::RS_UP_THETA];
+    const double& q1 = qVec[SIdx::RS_DOWN_THETA];
+    qr = (q0 - q1) / (2.0 * drs) - q;
+  }
+  // QAdder derivative with respect to the degeneracy parameter
+  double qt;
+  {
+    const std::vector<double> theta = structProp.getDegeneracyParameters();
+    const double dt = theta[SIdx::RS_THETA_UP] - theta[SIdx::RS_THETA];
+    const double q0 = qVec[SIdx::RS_THETA_UP] / rs[SIdx::RS_THETA];
+    const double q1 = qVec[SIdx::RS_THETA_DOWN] / rs[SIdx::RS_THETA];
+    qt = theta[SIdx::RS_THETA] * (q0 - q1) / (2.0 * dt);
+  }
+  return vector<double>({q, qr, qt});
+}
+
+// -----------------------------------------------------------------
+// qStructProp class
+// -----------------------------------------------------------------
+
+// Compute structural properties
+int QStructProp::compute() {
+  try {
+    if (!csrIsInitialized) {
+      int counter = 0;
+    for (size_t i = 0; i < csr.size(); ++i) {
+      const auto& in = csr[i].in;
+      auto& c = csr[i];
+      // If there was no fixed file specified in input
+      if (in.getFixed().empty() && i % 3 == 0){
+      // Initialize the csr objects with theta-dtheta, theta, theta+dtheta in order 
+      // to compute adrFixed if no file was specified in input.
+        c.init();
+        switch (counter % 3) { 
```

Comment:
I am not sure that we want this extra print statements, they clutter the code and the output. If you think that they are useful when running just let me know

Reply:
I added the print statements because they are only shown if fixedAdrQvs.py is run. When I was running the file to obtain the fixed components I wasn't sure when the next dtheta was computed that's basically why I added them. If SolveQVSStls.py is run then they aren't printed.

Reply:
Ok, if you think they are useful leave them then